core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/core/cis_lmsless/cis_lmsless.module

```diff
@@ -584,7 +584,7 @@ function _cis_lmsless_theme_vars() {
       if (isset($_SESSION['masquerading'])) {
         $vars['masquerade_logout'] = '<lrnsys-button label="' . t('revert (@name)', array('@name' => $GLOBALS['user']->name)) . '" href="' . url('masquerade/unswitch', array('query' => array('destination' => current_path(), 'token' => drupal_get_token('masquerade/unswitch')))) . '" icon="supervisor-account" hover-class="' . $vars['lmsless_classes'][$vars['distro']]['color'] . ' ' . $vars['lmsless_classes'][$vars['distro']]['dark'] . ' white-text"></lrnsys-button>';
       }
-      else {
+      else if(_cis_connector_role_groupings(array('staff','teacher'))) {
```

Comment:
doesn't this then remove the ability to "un-masquerade" if this is in an elseif?

Reply:
Nah, the un-masquerade button is in the if/isset section which runs before this. If the user is already masquerading, it won't check whether they're a student/teacher role group but I think that's how it worked before anyway.

So, if the user is already masquerading they get the "Revert" button from Line 585. If they're not already masquerading, and not in the staff/teacher role group, they won't be shown the form to masquerade at all.